apps/files/lib/Controller/ApiController.php

```diff
@@ -267,12 +268,9 @@ public function getRecentFiles() {
 	 * @param string $mode
 	 * @param string $direction
 	 * @return Response
-	 * @throws \OCP\PreConditionNotMetException
 	 */
-	public function updateFileSorting($mode, $direction) {
-		$allowedMode = ['name', 'size', 'mtime'];
-		$allowedDirection = ['asc', 'desc'];
-		if (!in_array($mode, $allowedMode) || !in_array($direction, $allowedDirection)) {
+	public function updateFileSorting($mode, $direction): Response {
+		if (!in_array($mode, Capabilities::SORTING_MODES) || !in_array($direction, Capabilities::SORTING_DIRECTIONS)) {
```

Comment:
```suggestion
		if (!in_array($mode, Capabilities::SORTING_MODES, true) || !in_array($direction, Capabilities::SORTING_DIRECTIONS, true)) {
```

for strict comparison